client/app/services/group/currentGroup.service.js

```diff
@@ -14,7 +14,18 @@ export default class CurrentGroup {
     Object.assign(this, {
       value: {},
       User,
-      SessionUser
+      SessionUser,
+      map: {
```

Comment:
So you basically mean not storing methods, but only storing data? Basically for the case that the parent scope is destroyed and the methods wouldn't do what they should anymore?
I don't really think it's an observer pattern. As far as I understood, this would mean something like having an `observers` array in our `groupMap` or `currentGroup`, that all other controllers could be part of / subscribe to, and our groupMap would alert all methods in this array on changes.
But since we don't have subscriptions to messages here, but only an interface to a function that's called in it's own scope, I'd also be fine with having a couple of watch patterns less, and keeping it like this. ...will read myself a little more into it though to really understand the problem. Maybe I can find something on that topic.